### PR TITLE
feat(sezm): add activation checkpoint option

### PR DESCRIPTION
**Pull request by OutisLi**
## Summary by CodeRabbit

* **New Features**
  * Introduced optional activation checkpointing during inference mode to enhance memory efficiency and performance on resource-limited systems.
  * Users can enable this optimization by setting the `DP_ACT_INFER=1` environment variable.
  * Automatically respects training state and gradient requirements for safe operation.

### CODE REVIEW

File: deepmd/pt/model/descriptor/sezm_nn/block.py [hunk 1]
```diff
@@ -543,6 +547,15 @@ def _extract_l0_from_canonical(self, value: torch.Tensor) -> torch.Tensor:
...
         return value[:, 0, :, :].reshape(value.shape[0], self.channels)
 
+    def _use_infer_activation_checkpoint(self, *tensors: torch.Tensor) -> bool:
+        """Return whether eval-time activation checkpointing should be used."""
+        return (
+            not self.training
+            and os.environ.get("DP_ACT_INFER") == "1"
+            and torch.is_grad_enabled()
+            and any(tensor.requires_grad for tensor in tensors)
+        )
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/pt/model/descriptor/sezm_nn/block.py:557]: The document will be organized latter altogether

File: deepmd/pt/model/descriptor/sezm_nn/block.py [hunk 2]
```diff
@@ -566,6 +579,31 @@ def _run_so2_unit(
...
+                lambda x_, radial_feat_: self._run_so2_unit_impl(
+                    x_,
+                    edge_cache_no_proj,
+                    radial_feat_,
+                ),
+                x,
+                radial_feat,
+                use_reentrant=False,
+                preserve_rng_state=True,
+            )
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply deepmd/pt/model/descriptor/sezm_nn/block.py:597]: Passed compile tests

**Review by wanghan-iapcm — COMMENTED** [review]
### Code review

Found 1 issue:

1. The new opt-in activation-checkpoint feature adds reachable branches (checkpoint-on vs. checkpoint-off in both `_run_so2_unit` and `_run_ffn_unit`, behind a four-condition guard) but ships no test. A unit test should exercise the `DP_ACT_INFER=1` path and assert the checkpointed output and gradients match the non-checkpointed path — the so2 branch also nulls `D_to_m_cache`/`Dt_from_m_cache`, so an equivalence test guards against silent divergence. Per the project testing convention, new features should cover every reachable code path and both branches of boolean conditions.

https://github.com/deepmodeling/deepmd-kit/blob/febea967eab7d6e2af19a5c15e2a6fe780ea5ed3/deepmd/pt/model/descriptor/sezm_nn/block.py#L549-L558
